collector/cluster_settings.go

```diff
@@ -21,22 +21,52 @@ import (
 	"net/url"
 	"path"
 	"strconv"
+	"strings"
 
+	"github.com/c2h5oh/datasize"
 	"github.com/go-kit/log"
 	"github.com/go-kit/log/level"
 	"github.com/imdario/mergo"
 	"github.com/prometheus/client_golang/prometheus"
 )
 
+type watermarkMetric struct {
+	Type          prometheus.ValueType
+	Desc          *prometheus.Desc
+	ValueWithUnit func(clusterSettings ClusterSettingsResponse) (float64, string)
+}
+
+func getUnitAndValue(value string) (float64, string) {
```

Comment:
I don't see any examples where the values have bytes, only percentage. Do you know which version this changed? 

It is against the best practices to have units be a label. Units should always be normalized (i.e. bytes or seconds). I think it would be best to keep the percentage here as a gauge, but convert percentage to a float from 0-1. 

I'm also not sure about bringing in another dependency to do the unit conversion.

Reply:
@sysadmind Hi, sorry for delay here. @tomachine prepared this for my issue. We aproach it this way, because this setting could be set both ways and this was sounds acceptable when he wrote this.

If you have other suggestion how to approach this, we are open to suggestions.

You can read about this in official docs and I think this was possible set to percent or absolute value in version 6. Link: https://www.elastic.co/guide/en/elasticsearch/reference/8.6/modules-cluster.html#disk-based-shard-allocation

> cluster.routing.allocation.disk.watermark.low logo cloud
    (Dynamic) Controls the low watermark for disk usage. It defaults to 85%, meaning that Elasticsearch will not allocate shards to nodes that have more than 85% disk used. It can alternatively be set to a ratio value, e.g., 0.85. It can also be set to an absolute byte value (like 500mb) to prevent Elasticsearch from allocating shards if less than the specified amount of space is available. This setting has no effect on the primary shards of newly-created indices but will prevent their replicas from being allocated.

Reply:
Reading the docs, I can see where this value may be set to a byte value. The docs also mention that the value could be a ratio like `0.8` so that should also be considered. 

Looking over this code again, I think overall it is probably pretty close to where it needs to be, however I'm not a fan of the added 3rd party dependency. The package does not seem to be widely used and it means this exporter has to depend on an outside party for another package.  I'm also not sure, based on the tests in that package that it would properly parse a value like `500mb` anyway.